frontend/controller/actions/identity.js

```diff
@@ -343,6 +343,15 @@ export default (sbp('sbp/selectors/register', {
       sbp('state/vuex/commit', 'login', loginAttributes)
       await sbp('chelonia/storeSecretKeys', () => transientSecretKeys)
 
+      if (!state) {
```

Comment:
Question about the comment above in line 335 (I can only comment on changed lines):

```js
      // TODO: This is temporary until username is no longer needed internally
      // in contracts
```

Is this still needed now that you went through and changed everything from usernames to identityContractIDs?

Reply:
That seems to be setting `username` if it's not been provided. I think doing this is fine but maybe the comment itself isn't (I don't remember the reason for it, but I had the intention of removing `username` from almost everywhere)